### PR TITLE
Add Hanabi seat order layout

### PR DESCRIPTION
**Pull request by ravihammond**
## Summary
- add Hanabi shuffle_player_order, defaulting to true
- expose seat-indexed state.seat_order while keeping agent keys stable
- preserve legacy deck mapping when shuffle_player_order=false

## Tests
- /Users/ravi/Documents/research/JaxMARL/.venv/bin/python -m pytest tests/hanabi/test_hanabi.py -q

### CODE REVIEW

File: jaxmarl/environments/hanabi/hanabi_game.py [hunk 1]
```diff
@@ -162,20 +166,24 @@ def reset_game(self, key: chex.PRNGKey) -> State:
...
         key, _key = jax.random.split(key)
         shuffled_pairs = jax.random.permutation(_key, color_rank_pairs, axis=0)
+        if self.shuffle_player_order:
+            seat_order = jax.random.permutation(key, self.agent_range)
+        else:
+            seat_order = self.agent_range
         # generate one-hot encoded deck
         deck = self._one_hot_encode_deck(shuffled_pairs)
-        return self.get_first_state(deck)
+        return self.get_first_state(deck, seat_order)
```
[the remark this answers is not in the document]
Comment (ravihammond) [reply jaxmarl/environments/hanabi/hanabi_game.py:186]: I think I have fixed this now, please review my latest commits.

File: jaxmarl/environments/hanabi/hanabi.py [hunk 2]
```diff
@@ -41,6 +42,7 @@ def __init__(
             max_info_tokens=max_info_tokens,
             max_life_tokens=max_life_tokens,
             num_cards_of_rank=num_cards_of_rank,
+            shuffle_player_order=shuffle_player_order,
         )
```
[the remark this answers is not in the document]
Comment (ravihammond) [reply jaxmarl/environments/hanabi/hanabi.py]: I think I have fixed this now, please review my latest commits.

File: tests/hanabi/test_hanabi.py [hunk 3]
```diff
@@ -111,6 +111,56 @@ def single_life_actions(env):
...
+    legal_moves = env.get_legal_moves(state)
+
+    assert env.agents == ["agent_0", "agent_1", "agent_2", "agent_3"]
+    assert list(obs.keys()) == env.agents
+    assert list(obs_again.keys()) == env.agents
+    assert list(legal_moves.keys()) == env.agents
+    assert list(env.action_spaces.keys()) == env.agents
+    assert jnp.array_equal(state.seat_order, state_again.seat_order)
+    assert jnp.array_equal(jnp.sort(state.seat_order), jnp.arange(env.num_agents))
+
```
[the remark this answers is not in the document]
Comment (ravihammond) [reply tests/hanabi/test_hanabi.py]: I think I have fixed this now, please review my latest commits.

**Comment by amacrutherford** [comment]
sounds good, gonna wait until #186 is in before looking at this :)

**Comment by amacrutherford** [comment]
@ravihammond a couple small edits including fixing the conflicts caused by #186. I have changed the default behaviour to `false` so current hanabi code behaves the same, shout if you think it should be `true`. I've also put these changes into the CHANGELOG as part of 0.2.0

**Review by amacrutherford — APPROVED** [review]
